src/data-workspace/inputs/generic-input.js

```diff
@@ -43,23 +43,51 @@ export const GenericInput = ({
         )
     }
 
+    const formatValue = (value) => {
+        let formattedValue = value
+
+        if (value && valueType === VALUE_TYPES.NUMBER) {
+            if (value.startsWith('.')) {
+                formattedValue = `0${value}`
+            }
+            if (value.length > 1 && value.endsWith('.')) {
+                formattedValue = value.slice(0, -1)
+            }
+            const unneededLeadingZero = /^0[0-9].*$/
+            if (unneededLeadingZero.test(value)) {
+                formattedValue = Number(value).toString()
+            }
+        }
+
+        return formattedValue
+    }
+
     const { input, meta } = useField(fieldname, {
         validate: validatorsByValueType[valueType],
         subscription: { value: true, dirty: true, valid: true },
+        format: formatValue,
+        formatOnBlur: true,
     })
 
     const handleBlur = () => {
         const { value } = input
         const hasEmptySpaces = value && value.trim() === ''
         const { dirty, valid } = meta
-        if (dirty && valid && !hasEmptySpaces && value !== lastSyncedValue) {
-            syncData(value.trim())
+        if (
+            value &&
```

Comment:
this seems to prevent the post of an empty value (i.e. a deletion of the data value)?

Reply:
Its something I found myself unexpected having to add this as a safety check for the use of `value.trim()` I was occasionally getting errors as value has `undefined` -  was a bit of a surprise as I had expect that `valid` would maybe deal with that but it didn't always seem to be the case.

Reply:
Looking at the previous behaviour here though it looks like it it should still submit if the field is empty so I'll see about changing this accordingly